Source/JavaScriptCore/runtime/Watchdog.cpp

```diff
@@ -60,7 +60,9 @@ void Watchdog::setTimeLimit(Seconds limit,
 bool Watchdog::shouldTerminate(JSGlobalObject* globalObject)
 {
     ASSERT(m_vm->currentThreadIsHoldingAPILock());
-    if (MonotonicTime::now() < m_deadline)
+
+    auto epsilon = Seconds::fromMilliseconds(20);
+    if (MonotonicTime::now() + epsilon < m_deadline)
```

Comment:
I think the use of an epsilon here deserves a bit of explanation for future reference.  Can you add a comment before landing please?

The tests in `testapi` are a bit fine tuned to minimize running time.  I think a 20ms might not matter, but I'm really not sure, and I wouldn't want `testapi` to become more flaky.  I would prefer that you conditionalize this change on `OS(WINDOWS)`, just so I know I won't have to debug potential spurious `testapi` failures.